### ISSUE TITLE
Download progress updates in the console / log / sonarr / radarr 

### ISSUE THREAD

**Issue by JustSuperHuman** [wouterdebie/putioarr#5]
Wondering if it would be reasonable to add a download percentage to the logs. 
```
[2023-12-25T22:20:01Z INFO ] Active transfers: 1 
                                      SomethingSomethingDownload (55%)
```

It would be really cool to see it in Sonarr / Radarr also. 

I have had a couple times where I can't tell if a file is stuck or is actually downloading.

**Comment by wouterdebie** [comment]
Yeah I actually implemented this by counting bytes while downloading, but
it slowed down the download significantly and I never got back to it and do
it properly.

The forwarding piece to sonarr/radarr is a different question though, since
right now I just pass on the download percentage from put.io, but in this
case there's obviously a second step where files are downloaded to local
storage. Sonarr/radarr don't make a difference between stages, but what I
could do is regard the put.io download as the first 50% and the download to
local as the other 50%. Implementing this is currently not trivial though,
since the different stages of the process share state with each other, so
I'd have to re-architect a bit.

I'll leave this open for now and might get to something in the next few
days.

**Comment by bugrax** [comment]
Opened #37 to address this.

It maps the two stages the *arr can't see separately onto its single progress bar:

- put.io cloud download → 0–50%
- pulling the files to local disk → 50–100%

Both stages move the same number of bytes, so the split is byte-accurate, and the transfer only reaches 100% once the files are actually on disk (so it doesn't trip the "No files found eligible for import" issue from #16). Orphaned watch-folder files have no cloud stage, so their local pull is reported directly as 0–100%.

To avoid the slowdown mentioned above, local progress is tracked with a lock-free `AtomicU64` counter incremented per streamed chunk and read only when the *arr polls or the status line is logged — nothing extra happens on the download hot path.

The periodic status log now shows the combined percentage too:

```
Active transfers: 2
  [abcd: Some.Movie.2024.1080p] (73%)
  [ef01: Some.Show.S01E04] (100%)
```

Tested on a live instance: the *arr progress bar now advances continuously through both stages instead of sitting at 0% during the local pull.
